src/store/index.js

```diff
@@ -9,17 +9,62 @@ import type { Data, Message } from './types';
 
 const store: Data = require('./emails.json');
 
-/**
- * You might find it useful to implement some functions that make accessing
- * deeply nested attributes of Messages or Threads...
- */
+function getNewMessageIds() {
+  return store.mailboxes.INBOX.threadIds;
+}
+
+function getNewMessages(newMessageIds) {
+  let newMessages = [];
+  for (let id = 0; id < newMessageIds.length; id += 1) {
+    const rawMessageObj = store.messages[newMessageIds[id]];
+    const refinedMessageObj = {
+      Snippet: rawMessageObj.snippet,
+      Date: '',
+      From: '',
+      To: '',
+      Subject: '',
+    };
+    for (let header of rawMessageObj.payload.headers) {
+      refinedMessageObj[header.name] = header.value;
+    };
+    newMessages.push(refinedMessageObj);
+  }
+  return newMessages;
+}
+
+function formatNewMessage(newMessage) {
+  const formattedRecipients = newMessage.To;
+  const formattedDate = formatTimestamp(newMessage.Date);
+  const formattedSender = newMessage.From.match(/([A-Z])\w+/g).join(' ');
+  const formattedSubject = formatSubject(newMessage.Subject);
+  const formattedSnippet = formatSnippet(newMessage.Snippet);
+  const formattedMessage = {
+    recipients: formattedRecipients,
+    sender: formattedSender,
+    subject: formattedSubject,
+    snippet: formattedSnippet,
+    date: formattedDate
+  }
+  return formattedMessage;
+}
+
+function formatTimestamp(datestring) {
+  const d = new Date(datestring);
+  const month = d.getMonth() + 1;
+  const day = d.getDay();
+  const year = String(d.getFullYear()).substring(2);
+  const timestamp = `${month}/${day}/${year}`;
+  return timestamp;
+}
+
+function formatSubject (subject) {
+  return subject.length > 33 ? subject.substring(0, 33) + '...' : subject;
+}
 
-// For example, this returns the subject of a given message if it is defined
-// by a header in the message payload
-export function getMessageSubject(message: Message): ?string {
-  const { headers } = message.payload;
-  const subjectHeader = headers.find(header => header.name === 'Subject');
-  return subjectHeader && subjectHeader.value;
+function formatSnippet (snippet) {
+  return snippet.length > 77 ? snippet.substring(0, 77) + '...' : snippet;
```

Comment:
Nice 👍

Reply:
Got it! I also wanna try this `...`, but have no idea. Learnt sth!